skrub/_minhash_encoder.py

```diff
@@ -9,19 +9,21 @@
 
 import numpy as np
 from joblib import Parallel, delayed, effective_n_jobs
-from numpy.typing import ArrayLike, NDArray
-from sklearn.base import BaseEstimator, TransformerMixin
+from numpy.typing import NDArray
+from sklearn.base import TransformerMixin
 from sklearn.utils import gen_even_slices, murmurhash3_32
-from sklearn.utils.validation import _check_feature_names_in, check_is_fitted
+from sklearn.utils.validation import check_is_fitted
 
+from . import _dataframe as sbd
 from ._fast_hash import ngram_min_hash
+from ._on_each_column import RejectColumn, SingleColumnTransformer
 from ._string_distances import get_unique_ngrams
-from ._utils import LRUDict, check_input
+from ._utils import LRUDict
 
 NoneType = type(None)
 
 
-class MinHashEncoder(TransformerMixin, BaseEstimator):
+class MinHashEncoder(SingleColumnTransformer, TransformerMixin):
```

Comment:
For the inheritance, the Mixin needs to be on the right of the `BaseEstimator`. Otherwise the tag do not work properly via the MRO.

Reply:
I'll open a PR to solve that because I probably merge some other PR with this issue.

Reply:
I'll change it here and in the gap encoder. Later we can consider removing the transformermixin because those transformers always output the correct dataframe type so have no need for the set_output stuff, and they are not used directly but always inside a tablevectorizer or similar